src/main/msp/msp.c

```diff
@@ -802,7 +802,9 @@ static bool mspCommonProcessOutCommand(uint8_t cmdMSP, sbuf_t *dst, mspPostProce
         osdFlags |= OSD_FLAGS_OSD_FEATURE;
 #endif
 #ifdef USE_MAX7456
-        osdFlags |= OSD_FLAGS_OSD_HARDWARE_MAX_7456;
+        if (max7456IsDeviceDetected()) {
+            osdFlags |= OSD_FLAGS_OSD_HARDWARE_MAX_7456;
```

Comment:
This should use a new flag - `OSD_FLAGS_OSD_HARDWARE_MAX_7456` is used to drive hiding / showing of all settings dependent on having a MAX7456 (powered or not) in configurator:

https://github.com/betaflight/betaflight-configurator/blob/53fb5944d14b9d66fd7fc7a47424c9ae075a26f4/src/js/tabs/osd.js#L1800

But the light of Unified Targets, we should probably make `OSD_FLAGS_OSD_HARDWARE_MAX_7456` dynamic, and dependent on `max7456Config->csTag && max7456Config->spiDevice`, to allow proper detection if a MAX7456 device is actually configured.

Reply:
I'm not sure I fully understand. In the Configurator the only logic that uses this flag (`haveMax7456Video`) is that which hides the font manager division. We don't seem to have any other logic based on actual detection of the max7456. Also `max7456IsDeviceDetected()` only returns try if the max7456 successfully completes detection and initialization. I don't see why checking for `max7456Config->csTag && max7456Config->spiDevice` is better.

Reply:
> In the Configurator the only logic that uses this flag (haveMax7456Video) is that which hides the font manager division.

That's incorrect - there are other parts of the configuration screen that are dependent on `requires-max7456`:
- video mode selection;
- active OSD profile selection (this is probably a bug though).

Requiring that a battery is plugged in on some flight controllers just to be able to configure the video format is contrary to doing this in the safest possible way.
Also, having a way to figure out in configurator that a MAX7456 is configured, but not powered will be helpful to display an appropriate notice to users re how they can flash fonts.

In general, there are 3 different possible configurations for MAX7456, and I think having two flags to encode them all makes sense:
- no MAX7456 configured;
- MAX7456 configured but not powered up;
- MAX7456 configured and powered up.